kitsune/wiki/tests/test_views.py

```diff
@@ -470,6 +471,98 @@ def test_needs_change(self):
         assert not doc.needs_change
         eq_('', doc.needs_change_comment)
 
+    def test_draft_revision_view(self):
+        """Test Draft Revision saving is working propoerly"""
+        rev = ApprovedRevisionFactory()
+        url = reverse('wiki.draft_revision', locale='bn-BD')
+        data = {
+            'content': 'Test Content bla bla bla',
+            'keyword': 'something',
+            'summary': 'some summary',
+            'title': 'test title',
+            'slug': 'test-slug',
+            'based_on': rev.id
+        }
+        resp = self.client.post(url, data)
+        eq_(201, resp.status_code)
+        obj = DraftRevision.objects.get(creator=self.u, document=rev.document, locale='bn-BD')
+        eq_(obj.content, data['content'])
+        eq_(obj.based_on, rev)
+
+    def test_draft_revision_many_times(self):
+        """Test only one draft revision for a single translation and user"""
+        rev = ApprovedRevisionFactory()
+        url = reverse('wiki.draft_revision', locale='bn-BD')
+        data = {
+            'content': 'Test Content bla bla bla',
+            'keyword': 'something',
+            'summary': 'some summary',
+            'title': 'test title',
+            'slug': 'test-slug',
+            'based_on': rev.id
+        }
+        # post 10 post request for draft
+        for i in range(10):
+            resp = self.client.post(url, data)
+            eq_(201, resp.status_code)
+
+        obj = DraftRevision.objects.filter(creator=self.u, document=rev.document, locale='bn-BD')
+        # There should be only one draft revision
+        eq_(1, obj.count())
+
+    def test_draft_revision_restore_in_translation_page(self):
+        """Check Draft Revision is restored when a user click on the Restore Button"""
+        # Create a draft revision
+        draft = DraftRevisionFactory(creator=self.u)
+        doc = draft.document
+        # Get the Translation page of the document
+        trans_url = reverse('wiki.translate', locale=draft.locale, args=[doc.slug])
+        trans_resp = self.client.get(trans_url)
+        eq_(200, trans_resp.status_code)
```

Comment:
I thinks getting the Translation page of the document is unneeded. The API to store the draft page should be stateless and not depend on the translation page being loaded first. Lines 519 through 521 should probably be removed.